run_all_unit_tests.sh

```diff
@@ -0,0 +1,132 @@
+#!/bin/bash
+source ~/.bashrc
+
+# check if current env already exist
+find_in_conda_env(){
+    conda env list | grep "${@}" >/dev/null 2>/dev/null
+}
+
+# error message when failed to lock
+lockfailed()
+{
+    echo "failed to get lock"
+    exit 1
+}
+
+# create an environment including the set of requirements specified in the requirements (if not already exist)
+create_env() {
+    force_cuda_version=$1
+    env_path=$2
+    mode=$3
+    start_time=`date +%s`
+
+    requirements=$(cat ./requirements/requirements.txt)
+    requirements+=$(cat ./requirements/requirements_dev.txt)
+
+    if [ $mode = "examples" ]; then
+        requirements+=$(cat ./fusedrug_examples/requirements.txt)
+    fi
+
+    # Python version
+    PYTHON_VER=3.8
+    ENV_NAME="fuse-drug_$PYTHON_VER-CUDA-$force_cuda_version-$(echo -n $requirements | sha256sum | awk '{print $1;}')"
+    echo $ENV_NAME
+
+    # env full name
+    if [ $env_path = "no" ]; then
+        env="-n $ENV_NAME"
+    else
+        env="-p $env_path/$ENV_NAME"
+    fi
+
+
+    # create a lock
+    mkdir -p ~/env_locks # will create dir if not exist
+    lock_filename=~/env_locks/.$ENV_NAME.lock
+    echo "Lock filename $lock_filename"
+
+    (
+        flock -w 1200 -e 873 || lockfailed # wait for lock at most 20 minutes
+
+        # Got lock - excute sensitive code
+        echo "Got lock: $ENV_NAME"
+
+        nvidia-smi
+
+        if find_in_conda_env $ENV_NAME ; then
+            echo "Environment exist: $env"
+        else
+            echo "Mode=$mode"
+            # create an environment
+            echo "Creating new environment: $env"
+            conda create $env python=$PYTHON_VER -y
+            echo "Creating new environment: $env - Done"
+
+            # install PyTorch
+            if [ $force_cuda_version != "no" ]; then
+                echo "forcing cudatoolkit $force_cuda_version"
+                conda install $env pytorch torchvision pytorch-cuda=$force_cuda_version -c pytorch -c nvidia
+                echo "forcing cudatoolkit $force_cuda_version - Done"
+            fi
+
+            echo "Installing FuseMedML"
+            conda run $env --no-capture-output --live-stream pip install git+https://github.com/BiomedSciAI/fuse-med-ml@master
```

Comment:
How do we force to bump fuse version? Change the requirement file?

Reply:
With that being said, if the conda env already exists we will NOT update fuse.

Reply:
Yes, I was asking about the case it already exists.